src/replay/trackers.zig

```diff
@@ -101,6 +101,25 @@ pub const SlotTracker = struct {
         }
         return frozen_slots;
     }
+
+    pub fn parents(
+        self: *const SlotTracker,
+        allocator: Allocator,
+        slot: Slot,
+    ) Allocator.Error![]const Slot {
+        var parents_list = std.ArrayListUnmanaged(Slot){};
+        errdefer parents_list.deinit(allocator);
+
+        var current_slot = slot;
+        while (self.slots.get(current_slot)) |current| {
+            const parent_slot = current.constants.parent_slot;
+            try parents_list.append(allocator, parent_slot);
```

Comment:
Question: do you think it would be possible to `ensureTotalCapacity` on `parents_list` with some reasonable estimate? Ie `try parents_list.ensureTotalCapacity(allocator, self.slots.count() - (self.slots.getIndex(slot) orelse self.slots.count()))`? Would still need to `try append`, but may reduce the allocation overhead in common cases.

Reply:
I think that should work. Something like this? [a86bcf1](https://github.com/Syndica/sig/pull/762/commits/a86bcf1104a903eb000a0af6e65095f29f449c90)

Reply:
> As in, will the distance of a slot relative to the end always represent the upper bound of the number of parents?

I think so, but not totally sure. To be on the safe side, I switch to `append` instead of `appendAssumeCapacity`